### PR TITLE
Migrate to JUnit 5

### PR DESCRIPTION
**Pull request by stevebosman**
I've updated a lot of the remaining tests to JUnit 5

I think this will leave the tests extending the following to do:

* `AbstractDateConverterTest`
* `AbstractNumberConverterTest`
* `BaseLocaleConverterTest`

### CODE REVIEW

**Review by garydgregory — CHANGES_REQUESTED** [review]
Hello @stevebosman 

Thank you for your PR. 

It will take a lot longer to review a PR when it contains unnecessary changes like editing public/protected/private. Also please keep the style to use static imports only for assertion methods. This will all reduce the noise in the change set a lot.

IMO a PR for this task is not about creating the prettiest code but creating a change set that easier for a maintainer to review, and there are 105 files changed here 😉

**Comment by stevebosman** [comment]
I've added the changes for static imports. I've also reverted the visibility changes. I think I've caught all of them, but will have another look later. My fault for trusting my IDE's refactoring too much :)

**Comment by garydgregory** [comment]
@stevebosman 

Thank you for your update. Yeah, all IDEs I've tried mess up the conversion or leaves you with code that does not even compile (I'm looking at you IntelliJ...)

Another item that would be nice to have (but don't feel obligated):
```
assertTrue(e.toString().contains("Unknown property"), "Wrong exception was thrown: " + e);
```
->
```
assertTrue(e.toString().contains("Unknown property"), () -> "Wrong exception was thrown: " + e);
```
which will avoid creating garbage Strings in the majority of build runs.

**Comment by stevebosman** [comment]
I have fixed that one and the few others in `BeanComparatorTestCase`. There are similar in some other test files, but my instinct is they should be addressed in another much smaller PR.

File: src/test/java/org/apache/commons/beanutils2/WrapDynaBeanTestCase.java [hunk 1]
```diff
@@ -266,10 +276,10 @@ public void testMappedRemove() {
         }
 
         try {
-            assertTrue("Can not see unknown key", !bean.contains("mappedProperty", "Unknown Key"));
+            assertFalse(bean.contains("mappedProperty", "Unknown Key"), "Can not see unknown key");
             bean.remove("mappedProperty", "Unknown Key");
             fail("Should have thrown UnsupportedOperationException");
-            // assertTrue("Can not see unknown key",
+            // Assert.assertTrue("Can not see unknown key",
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/beanutils2/WrapDynaBeanTestCase.java:282]: Odd comment change ;-)

File: src/test/java/org/apache/commons/beanutils2/bugs/Jira157TestCase.java [hunk 2]
```diff
@@ -89,16 +84,16 @@ protected void setUp() throws Exception {
      *
      * @throws Exception
      */
-    @Override
+    @AfterEach
     protected void tearDown() throws Exception {
-        super.tearDown();
     }
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/beanutils2/bugs/Jira157TestCase.java:89]: So this method can be removed?

File: src/test/java/org/apache/commons/beanutils2/bugs/Jira273TestCase.java [hunk 3]
```diff
@@ -16,54 +16,48 @@
...
-
     /**
      * Sets up.
      *
      * @throws Exception
      */
-    @Override
+    @BeforeEach
     protected void setUp() throws Exception {
-        super.setUp();
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/beanutils2/bugs/Jira273TestCase.java:51]: Remove method?

File: src/test/java/org/apache/commons/beanutils2/bugs/Jira273TestCase.java [hunk 4]
```diff
@@ -16,54 +16,48 @@
...
 
     /**
      * Tear Down.
      *
      * @throws Exception
      */
-    @Override
+    @AfterEach
     protected void tearDown() throws Exception {
-        super.tearDown();
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/beanutils2/bugs/Jira273TestCase.java:61]: Remove method?

File: src/test/java/org/apache/commons/beanutils2/bugs/Jira298TestCase.java [hunk 5]
```diff
@@ -16,56 +16,50 @@
...
-
     /**
      * Sets up.
      *
      * @throws Exception
      */
-    @Override
+    @BeforeEach
     protected void setUp() throws Exception {
-        super.setUp();
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/beanutils2/bugs/Jira298TestCase.java:53]: Remove method?

File: src/test/java/org/apache/commons/beanutils2/bugs/Jira298TestCase.java [hunk 6]
```diff
@@ -16,56 +16,50 @@
...
 
     /**
      * Tear Down.
      *
      * @throws Exception
      */
-    @Override
+    @AfterEach
     protected void tearDown() throws Exception {
-        super.tearDown();
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/beanutils2/bugs/Jira298TestCase.java:63]: Remove method?

**Review by garydgregory — APPROVED** [review]
LGTM, I made a few notes, but I'll merge as is.

**Pull request merged**